enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+* The cluster default primary network defined today will remain in place as the default network pods attach to. The cluster
+default network will continue to serve as the primary network for pods in a namespace that has no user-defined network. Pods
+with user-defined networks will still attach to the cluster default network with limited access to Kubernetes system resources.
+Pods with user-defined networks will have at least two network interfaces, one connected to the cluster default network and one
+connected to the user-defined network.
+* Allow multiple namespaces per network.
+* Support cluster ingress/egress traffic for user-defined networks, including secondary networks.
+* Support for ingress/egress features where possible:
+    * EgressQoS
+    * EgressService
+    * EgressIP
+    * Load Balancer and NodePort Services, as well as services with External IPs.
+* In addition to ingress service support, there will be support for Kubernetes services in user-defined networks. The
+scope of reachability to that service as well as endpoints selected for that service will be confined to the network
+and corresponding namespace(s) where that service was created.
+* Support for pods to continue to have access to the cluster default primary network for DNS and KAPI service access.
+* Kubelet healthchecks/probes will still work on all pods.
+* OpenShift Router/Ingress will work with some limitations for user-defined networks.
+
+### Non-Goals
+
+* Allowing different service CIDRs to be used in different networks.
+* Localnet will not be supported initially for primary networks.
+* Allowing multiple primary networks per namespace.
+* Hybrid overlay support on user-defined networks.
+
+### Future-Goals
+
+* DNS lookup for pods returning records for IPs on the user-defined network. In the first phase DNS will return the pod
+IP on the cluster default network instead.
+* Admin ability to configure networks to have access to all services and/or expose services to be accessible from all
+networks.
+* Ability to advertise user-defined networks to external networks using BGP/EVPN. This will enable things like:
+    * External -> Pod ingress per VRF (Ingress directly to pod IP)
+    * Multiple External Gateway (MEG) in a BGP context, with ECMP routes
+* Allow connection of multiple networks via explicit router API configuration.
+
+## Proposal
+
+By default in OVN-Kubernetes, pods are attached to what is known as the “cluster default" network, which is a routed network
+divided up into a subnet per node. All pods will continue to have an attachment to this network, even when assigned a
+different primary network. Therefore, when a pod is assigned to a user-defined network, it will have two interfaces, one
+to the cluster default network, and one to the user-defined network. The cluster default network is required in order to provide:
+
+1. KAPI service access
+2. DNS service access
+3. Kubelet healthcheck probes to the pod
```

Comment:
hmm, good point. We want to support this.

Reply:
I think to support it we need to allow a user to indicate on the pod which ports they want opened. I'm not sure what the best way is to do that yet. It would make sense on a per pod basis. We could probably leverage the prometheus annotations:

```
apiVersion: v1
kind: Pod
metadata:
  name: my-app-pod
  annotations:
    prometheus.io/scrape: "true"
    prometheus.io/path: /metrics
    prometheus.io/port: "8080"
```

or if that use case is too narrow, we can come up with some generic annotation to allow access to additional ports on the pod via the cluster default network.